src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -172,21 +173,39 @@ internal static void ValidateNumber(ReadOnlySpan<byte> utf8FormattedNumber)
                 return;
             }
 
+            //The non digit character inside the number
             byte val = utf8FormattedNumber[i];
 
             if (val == '.')
             {
                 i++;
+
+                while (i < utf8FormattedNumber.Length && JsonHelpers.IsDigit(utf8FormattedNumber[i]))
+                {
+                    i++;
+                }
+
+                if (utf8FormattedNumber.Length < i)
+                {
+                    throw new ArgumentException(SR.RequiredDigitNotFoundEndOfData, nameof(utf8FormattedNumber));
+                }
+            }
+
+
```

Comment:
Please remove the second of the two blank lines.  (Or, if you prefer, the first... not like I can tell :))